src/nvim/os/env.c

```diff
@@ -456,6 +456,15 @@ char *vim_getenv(const char *name)
     if (p_hf != NULL && vim_strchr(p_hf, '$') == NULL) {
       vim_path = (char *)p_hf;
     }
+
+#ifdef WIN32
+    char exe_name[MAXPATHL];
+    if (vim_path == NULL) {
+      GetModuleFileName(NULL, exe_name, MAXPATHL);
+      vim_path = exe_name;
```

Comment:
In Vim: https://github.com/vim/vim/blob/ac360bf2ca293735fc7c6654dc2b3066f4c62488/src/os_win32.c#L357 it mentions that "Maximum length of $PATH is more than MAXPATHL"--around 10x more. But MSDN uses MAXPATHL so whatever.

Anyways libuv `uv_exepath` function looks useful, and it avoids problems with Unicode paths. There is `GetModuleFileNameA` (ASCII) and `GetModuleFileNameW`, but we are calling `GetModuleFileName`, what does that do? libuv takes care of it for us.